plugin/build.gradle.kts

```diff
@@ -195,6 +197,11 @@ tasks.withType<Test> {
             maxFailures.set(10)
         }
     }
+
+    // AGP requires java 17, so set it to 17 for tests
+    javaLauncher = javaToolchains.launcherFor {
+        languageVersion = JavaLanguageVersion.of(17)
```

Comment:
I think that makes it easy to miss these tests on local builds. maybe instead of matrixing on github actions, it would be better to define separate test tasks with different toolchain setups to cover the different jvm versions?

Reply:
or maybe a separate source set just for the jdk17 tests, and then let the other tests run without toolchain and be matrixed by the github actions as normal?

Reply:
nevermind, that wasn't as clean as I thought it would be. working on something closer to your suggestion now...